packages/labs/eleventy-plugin-lit/src/test/eleventy-plugin-test.ts

```diff
@@ -0,0 +1,183 @@
+
+  async setup() {
+    // Simulate installing @lit-labs/eleventy-plugin with a symlink to the root
+    // of this package.
+    await fs.mkdir(pathlib.join(this._tempDir, 'node_modules', '@lit-labs'), {
+      recursive: true,
+    });
+    await fs.symlink(
+      '../../../../..',
+      pathlib.join(
+        this._tempDir,
+        'node_modules',
+        '@lit-labs',
+        'eleventy-plugin'
+      )
+    );
+  }
+
+  async cleanup() {
+    await fs.rm(this._tempDir, {recursive: true});
+  }
+
+  async write(files: {[path: string]: string}) {
+    await Promise.all(
+      Object.entries(files).map(async ([relative, data]) => {
+        const absolute = pathlib.resolve(this._tempDir, relative);
+        await fs.mkdir(pathlib.dirname(absolute), {recursive: true});
+        return fs.writeFile(absolute, normalize(data), 'utf8');
+      })
+    );
+  }
+
+  async read(path: string): Promise<string> {
+    const absolute = pathlib.resolve(this._tempDir, path);
+    return normalize(await fs.readFile(absolute, 'utf8'));
+  }
+
+  exec(command: string): Promise<{code: number}> {
+    const cwd = pathlib.resolve(this._tempDir);
+    const child = spawn(command, [], {
+      cwd,
+      shell: true,
+      stdio: 'inherit',
+    });
+    return new Promise((resolve) => {
+      child.on('exit', (code) => {
+        // Code will be null when the process was killed via a signal. 130 is
+        // the conventional return code used to represent this case.
+        resolve({code: code ?? 130});
+      });
+    });
+  }
+}
+
+const test = suite<{rig: TestRig}>();
+
+test.before.each(async (ctx) => {
+  ctx.rig = new TestRig();
+  await ctx.rig.setup();
+});
+
+test.after.each(async ({rig}) => {
+  await rig.cleanup();
+});
+
+test('without plugin', async ({rig}) => {
+  await rig.write({
+    'index.md': `
+      # Heading
+      <my-element></my-element>
+    `,
+  });
+  assert.equal((await rig.exec('eleventy')).code, 0);
+  assert.equal(
+    await rig.read('_site/index.html'),
+    normalize(`
+      <h1>Heading</h1>
+      <p><my-element></my-element></p>
+    `)
+  );
+});
+
+test('basic component', async ({rig}) => {
+  await rig.write({
+    // md
+    'index.md': `
+      # Heading
+      <my-element></my-element>
+    `,
+
+    // config
+    '.eleventy.cjs': `
+      const litPlugin = require('@lit-labs/eleventy-plugin');
+      module.exports = function (eleventyConfig) {
+        eleventyConfig.addPlugin(litPlugin, {
+          componentModules: ['./js/my-element.js'],
+        });
+      };
+    `,
+
+    // js
+    'js/my-element.js': `
+      import { html, LitElement } from 'lit';
+      class MyElement extends LitElement {
+        render() {
+          return html\`<b>shadow content</b>\`;
+        }
+      }
+      customElements.define('my-element', MyElement);
+    `,
+  });
+  assert.equal((await rig.exec('eleventy --config .eleventy.cjs')).code, 0);
```

Comment:
amazing test rig! 🎉

Reply:
Thanks! I like being able to see all the files defined together within the test function, instead of having to deal with `testdata/` directories which is the other way we could do this.

Only real downside is not having syntax highlighting. Could have a no-op `html` tag for HTML files, but (at least in my VSCode setup) there is no support for `js` or `md` tag highlighting.